File: programming-languages/java/trie/src/main/java/com/ac/MainTrie.java

package com.ac;

public class MainTrie {


    static void main() {
        Trie trie = new Trie();
        trie.insert("apple");
        trie.insert("app");
        trie.insert("airbus");
        trie.insert("access");
        trie.insert("air");
        System.out.println(trie);

        System.out.println(trie.findStartsWith("app"));
    }

}

File: programming-languages/nim/trie/src/maintrie.nim
import trie

proc main*() =
  var trie = newTrie()

  trie.insert("apple")
  trie.insert("app")
  trie.insert("banana")

  echo trie.search("apple")
  echo trie.search("ap")
  echo trie.startsWith("app")
  echo trie.findStartsWith("app")

  echo trie

when isMainModule:
  main()
